pkgs/development/libraries/rocthrust/default.nix

```diff
@@ -0,0 +1,91 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, rocm-cmake
+, rocm-runtime
+, rocm-device-libs
+, rocm-comgr
+, rocprim
+, hip
+, gtest ? null
+, buildTests ? false
+, buildBenchmarks ? false
+, rocmVersion ? "5.3.1" # rocThrust depends on specific ROCm versions per-release
```

Comment:
IMO just reference rocm's version directly. IOW: use `hip.version` and then users can just override the `hip` input.

Reply:
The problem is if we do that, then we'll update all the packages which use ``hip.version`` inadvertently.
While theoretically that could be fine, it could lead to strange bugs and other package breakage.
That's why I set ``rocmVersion`` and make sure that ``rocmVersion`` is equal to ``hip.version``, so that we don't inadvertently update packages in an impure way and also ensure that we are using hip with the same rocm version as the package itself.
Would a better alternative to be ``broken = (rocmVersion != hip.version)``?

Reply:
I think that's the best solution.
I'll get to it right after I'm done with ``composable_kernel``.